### PR TITLE
[Intel GPU] Support RegisterSparseXPU.cpp codegen.

### PR DESCRIPTION
**Pull request by xiaowangintel**
This PR is to support code generation for sparse operations on Intel GPUs.

### CODE REVIEW

File: torchgen/model.py [hunk 1]
```diff
@@ -279,6 +279,7 @@ def codegen_per_backend_entries() -> str:
     DispatchKey.CUDA,
     DispatchKey.MPS,
     DispatchKey.XPU,
+    DispatchKey.SparseXPU,
```
Comment (EikanWang) [inline torchgen/model.py:282]: @ZhiweiYan-96 , should we add comprehensive XPU dispatch keys here?

**Comment by EikanWang** [comment]
@xiaowangintel , pls. correct the github id of the commit.

**Comment by xiaowangintel** [comment]
> @xiaowangintel , pls. correct the github id of the commit.

Done

**Comment by EikanWang** [comment]
@xiaowangintel , pls. sign the CLA.

File: torchgen/model.py (hunk 1 above)
Comment (ZhiweiYan-96) [reply torchgen/model.py:282]: Yes, but weneed confirm whether `RegisterSparseXPU.cpp` works. We need add codes to cmake and make the object `RegisterSparseXPU.o`  link to corresponding library.

**Review by EikanWang — APPROVED** [review]

File: torchgen/model.py (hunk 1 above)
Comment (xiaowangintel) [reply torchgen/model.py:282]: Yeah, these codes have been added and tested on torch-xpu-ops. You can read https://github.com/intel/torch-xpu-ops/pull/1030/files.

**Review by atalman — APPROVED** [review]
lgtm

**Pull request closed without merging**